SingularityService/src/main/java/com/hubspot/singularity/resources/TaskResource.java

```diff
@@ -609,4 +613,30 @@ private SingularityTaskShellCommandRequest startShellCommand(SingularityTaskId t
     SingularityTaskId taskIdObj = getTaskIdFromStr(taskId);
     return taskManager.getTaskShellCommandUpdates(new SingularityTaskShellCommandRequestId(taskIdObj, commandName, commandTimestamp));
   }
+
+  @GET
+  @Path("/download/{httpPrefix}/{slaveHostname}/port/{port}/path/{path}")
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  @Operation(summary = "Proxy a file download from a Mesos Slave through Singularity")
+  @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+  public Response downloadFileOverProxy(
+      @Parameter(required = true, description = "Https (if available) or http") @PathParam("httpPrefix") String httpPrefix,
+      @Parameter(required = true, description = "Mesos slave hostname") @PathParam("slaveHostname") String slaveHostname,
+      @Parameter(required = true, description = "Http port for Mesos slave") @PathParam("port") String httpPort,
+      @Parameter(required = true, description = "Full file path to file on Mesos slave to be downloaded") @PathParam("path") String fileFullPath
+  ) {
+
+    Client client = ClientBuilder.newClient();
```

Comment:
I believe we want to inject & use the `AsyncHttpClient` here instead of using a JAX-RS client, similarly to https://github.com/HubSpot/Singularity/blob/c2d88711176b436a0e1683a83953222b7be0443d/SingularityService/src/main/java/com/hubspot/singularity/data/SandboxManager.java#L58-L63.

Reply:
Ah neat, will do. We already have one being injected too which is awesome since creating these clients are expensive operations.